File: localactorref.go
package gam

type LocalActorRef struct {
	mailbox Mailbox
}

func NewLocalActorRef(mailbox Mailbox) *LocalActorRef {
	return &LocalActorRef{
		mailbox: mailbox,
	}
}

func (ref *LocalActorRef) Tell(message interface{}) {
	ref.mailbox.PostUserMessage(message)
}

func (ref *LocalActorRef) SendSystemMessage(message SystemMessage) {
	ref.mailbox.PostSystemMessage(message)
}

func (ref *LocalActorRef) Stop() {
	ref.SendSystemMessage(&Stop{})
}

func (ref *LocalActorRef) Suspend() {
	ref.mailbox.Suspend()
}

func (ref *LocalActorRef) Resume() {
	ref.mailbox.Resume()
}

File: languages/csharp/GAM.Actor/LocalActorRef.cs
namespace GAM
{
    public class LocalActorRef : ActorRef
    {
        public IMailbox Mailbox { get;  }
        public LocalActorRef(IMailbox mailbox)
        {
            Mailbox = mailbox;
        }

        public override void Tell(object message)
        {
            Mailbox.PostUserMessage(message);
        }
    }
}